From db79b1540cb8cc442289e145170d450d8c1f2798 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Thu, 29 Aug 2019 14:29:46 +0800
Subject: [PATCH] drm: bridge: synopsys: dw-hdmi: Set dw-hdmi driver data

If hdmi driver data isn't be set, hdmi pointer can't be get
correctly, system will crush when reboot or shutdown.

Change-Id: I3a17a3238c307ab28d7570e8803a6dbe8c57c930
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
index d3cdb52eeced..2e1ae88a90ba 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -3848,6 +3848,7 @@ __dw_hdmi_probe(struct platform_device *pdev,
 	if (hdmi->i2c)
 		dw_hdmi_i2c_init(hdmi);
 
+	dev_set_drvdata(dev, hdmi);
 	dw_hdmi_register_debugfs(dev, hdmi);
 
 	if (of_property_read_bool(np, "hdcp1x-enable"))
-- 
2.35.3

